### PR TITLE
Fix: Retain focus for persisted input elements during view transitions

### PR DESCRIPTION
**Pull request by martrapp**
## Changes

An element with `transition:persist` survives view transitions and continues its life on the next page. Its state is preserved. For example, a `HTMLTextAreaElement` retains the text entered up to that point. Unfortunately, persisted elements lose focus during the transition when they are removed from the DOM and reinserted.

This fix saves the current focus and cursor position and restores them on the new page.    

## Testing

Added e2e test

## Docs

n/a bug fix

### CODE REVIEW

File: packages/astro/e2e/fixtures/view-transitions/astro.config.mjs
Comment (martrapp) [inline packages/astro/e2e/fixtures/view-transitions/astro.config.mjs:1]: Not really related to focus handling, but as I was going through the tests I noticed that we are running with `output = server`, when in fact almost all tests are static. So I marked the test that needs the server with `prerender = false` and changed to `hybrid`.

File: packages/astro/e2e/fixtures/view-transitions/src/components/Layout.astro
Comment (martrapp) [inline packages/astro/e2e/fixtures/view-transitions/src/components/Layout.astro:1]: The file was always called `styles.css`, but I noticed it only now

File: packages/astro/e2e/view-transitions.test.js [hunk 1]
```diff
@@ -788,7 +788,7 @@ test.describe('View Transitions', () => {
 
 	test('replace history', async ({ page, astro }) => {
 		await page.goto(astro.resolveUrl('/one'));
-		// page six loads the router and automatically uses the router to navigate to page 1
```
Comment (martrapp) [inline packages/astro/e2e/view-transitions.test.js:791]: copy & paste error from another test

**Review by natemoo-re — APPROVED** [review]
This seems great, thanks for the fix and detailed test!

**Pull request merged**